python/samba/dbchecker.py

```diff
@@ -2306,7 +2307,7 @@ def check_object(self, dn, attrs=['*']):
                 self.fix_metadata(obj, att)
 
         if self.is_fsmo_role(dn):
-            if "fSMORoleOwner" not in obj and ("*" in attrs or "fsmoroleowner" in map(str.lower, attrs)):
+            if "fSMORoleOwner" not in obj and ("*" in attrs or "fsmoroleowner" in attrs_lower):
                 self.err_no_fsmoRoleOwner(obj)
                 error_count += 1
 
```

Comment:
hmm, while I don't disaggree with this change, AFAICS the change (list/map) for dbchecker,py above actually isn't necessary for py2/py3. It looks to me this code will run fine as is (under python2/python3) If I am correct then this change probably should be in a separate patchset right?

Reply:
My reasons:
1. don't repeat yourself
2. use `in` on a map iter has a risk:
```
x = map(str, range(1,2,3))
'1' in x   # first time, return True
'1' in x   # second time, return False
```
In this case it won't happen, but the new code is more robust.

Reply:
like I said, I don't have a problem with the change, It should however be in a separate commit (and/or patchset) as it isn't really a py2/py3 compatibility change (and isn't necesary for the py3 porting) It shouldn't be intermingled with the py3 porting changes, this change is imho independent (and yes... useful/good/etc)

Reply:
OK, removed.